File: CPP-164/chapter.9/lab9.25/ShoppingCartManager.cpp
#include <iostream>
#include "ShoppingCart.cpp"

using namespace std;

void printMenu(ShoppingCart cart);
void menuReply(ShoppingCart cart);

int main(int argc, char const *argv[])
{
    string name, date;
    printf("Enter Customer's Name:\n");
    getline(cin, name);
    printf("Enter Today's Date:\n");
    getline(cin, date);

    ShoppingCart cart(name, date);
    printf("\nCustomers Name: %s\n", cart.getCustomerName().c_str());
    printf("Today's Date: %s\n", cart.getDate().c_str());
    printf("\n");
    cart.getCustomerName();
    cart.getDate();
    printMenu(cart);
    return 0;
}

void printMenu(ShoppingCart cart)
{
    printf("\nMENU\n");
    printf("a - Add item to cart\n");
    printf("d - Remove item from cart\n");
    printf("c - Change item from cart\n");
    printf("i - Output items' descriptions\n");
    printf("o - Output shopping cart\n");
    printf("q - Quit\n");
    printf("\n");
    menuReply(cart);
}

void menuReply(ShoppingCart cart)
{
    ItemToPurchase item;
    char ch;
    string itemName, itemDescription;
    int itemPrice, itemQuantity;

    cin >> ch;
    cin.ignore(32767, '\n');
    cin.clear();
    switch (ch)
    {
    case 'a':
        printf("ADD ITEM TO CART\n");
        printf("Enter the item name:\n");
        getline(cin, itemName);
        printf("Enter the item description:\n");
        getline(cin, itemDescription);
        printf("Enter the item price:\n");
        cin >> itemPrice;
        printf("Enter the item quantity:\n");
        cin >> itemQuantity;

        item.setName(itemName);
        item.setDescription(itemDescription);
        item.setPrice(itemPrice);
        item.setQuantity(itemQuantity);

        cart.addItem(item);   
        printMenu(cart);
        break;
        // needs fixing 
    // case 'd':
        // needs fixing
    // case 'c':
    //     printf("CHANGE ITEM QUANTITY\n");
    //     printf("Enter the item name:\n");
    //     getline(cin, itemName);
    //     printf("Enter the new quantity:\n");
    //     cin >> itemQuantity;
    //     cart.modifyItem(itemName, itemQuantity);
    //     printMenu(cart);
    //     break;
    case 'i':
        printf("OUTPUT ITEMS' DESCRIPTIONS\n");
        cart.printDescriptions();
        printMenu(cart);
        break;
        // fix printing out default constructor (none 0 @ 0.....)
    case 'o':
        printf("OUTPUT SHOPPING CART\n");
        cart.printTotal();
        printMenu(cart);
        break;
    case 'q':
        exit(0);
        break;
    default:
        break;
    }
}

File: Java-164/chapter.9/lab9.25/ShoppingCartManager.java
import java.util.Scanner;

public class ShoppingCartManager {
    public static Scanner scnr = new Scanner(System.in);

    public static void main(String[] args) {
        System.out.println("Enter Customer's Name:");
        String name = scnr.nextLine();
        System.out.println("Enter Today's Date:");
        String date = scnr.nextLine();

        System.out.println();
        ShoppingCart cart = new ShoppingCart(name,date);
        System.out.printf("Customer Name: %s\n",cart.getCustomerName());
        System.out.printf("Today's Date: %s\n",cart.getDate());
        cart.getCustomerName();
        cart.getDate();
        printMenu(cart);
    }

    public static void printMenu(ShoppingCart cart) {
        System.out.println("\nMENU");
        System.out.println("a - Add item to cart");
        System.out.println("d - Remove item from cart");
        System.out.println("c - Change item quantity");
        System.out.println("i - Output items' descriptions");
        System.out.println("o - Output shopping cart");
        System.out.println("q - Quit");
        System.out.println();
        menuReply(cart);
    }

    public static void menuReply(ShoppingCart cart) {
        ItemToPurchase item = new ItemToPurchase();
        System.out.println("Choose an option:");
        String ch = scnr.next();
        switch (ch) {
            case "a":
                System.out.println("ADD ITEM TO CART");
                scnr.nextLine();
                System.out.println("Enter the item name:");
                item.setName(scnr.nextLine());
                System.out.println("Enter the item description:");
                item.setDescription(scnr.nextLine());
                System.out.println("Enter the item price:");
                item.setPrice(scnr.nextInt());
                System.out.println("Enter the item quantity:");
                item.setQuantity(scnr.nextInt());
                cart.addItem(item);
                printMenu(cart);
                break;
            case "d":
                System.out.println("REMOVE ITEM FROM CART");
                scnr.nextLine();
                System.out.println("Enter name of item to remove:");
                cart.removeItem(scnr.nextLine());
                printMenu(cart);
                break;
            case "c":
                System.out.println("CHANGE ITEM QUANTITY");
                scnr.nextLine();
                System.out.println("Enter the item name:");
                String name = scnr.nextLine();
                System.out.println("Enter the new quantity:");
                int quantity = scnr.nextInt();
                cart.modifyItem(name, quantity);
                printMenu(cart);
                break;
            case "i":
                System.out.println("OUTPUT ITEMS' DESCRIPTIONS");
                cart.printDescriptions();
                printMenu(cart);
                break;
            case "o":
                System.out.println("OUTPUT SHOPPING CART");
                cart.printTotal();
                printMenu(cart);
                break;
            case "q":
                break;
            default:
                menuReply(cart);
                break;
        }
    }
}